core/src/Cuda/Kokkos_Cuda_Parallel_Team.hpp

```diff
@@ -711,16 +711,17 @@ class ParallelReduce<CombinedFunctorReducerType,
     }
   }
 
-  __device__ inline void run(SHMEMReductionTag&, const int& threadid) const {
+  __device__ inline void run(SHMEMReductionTag, const int& threadid) const {
     const integral_nonzero_constant<word_size_type,
                                     ReducerType::static_value_size() /
                                         sizeof(word_size_type)>
         word_count(m_functor_reducer.get_reducer().value_size() /
                    sizeof(word_size_type));
 
-    reference_type value = m_functor_reducer.get_reducer().init(
-        kokkos_impl_cuda_shared_memory<word_size_type>() +
-        threadIdx.y * word_count.value);
+    reference_type value =
+        m_functor_reducer.get_reducer().init(reinterpret_cast<pointer_type>(
```

Comment:
So it is not so much a "adding test", the code was broken wasn't it?

Reply:
Well, yeah, it was broken for `Cuda` and `HIP` but not for the other backends AFAIK which means we didn't have a test.